kernel/src/client.rs

```diff
@@ -0,0 +1,128 @@
+use crate::prelude::*;
+use crate::types::Value;
+use common::message::{serialize, MessageParser, MessageSerializer, Messages, ReplyMessage};
+use common::ringbuffer::RingBuffer;
+
+extern crate alloc;
+use alloc::boxed::Box;
+use alloc::collections::VecDeque;
+use alloc::sync::Arc;
+use alloc::vec::Vec;
+
+pub struct Client<'a> {
```

Comment:
I don't understand what the point of this struct is, it seems to mostly be the single function `process_incoming_message`, and doesn't really encapsulate anything.